src/content/turbosnap/turbosnap-dependency-tracing.mdx

```diff
@@ -0,0 +1,137 @@
+  "name": "./src/inputs/PinInput/stories/PinInput.stories.tsx + 4 modules",
+  "modules": [
+    { "name": "./src/inputs/PinInput/stories/PinInput.stories.tsx" },
+    {
+      "name": "./src/inputs/PinInput/stories/PinInputWithNoCopyPaste.storyfile.tsx"
+    },
+    {
+      "name": "./src/inputs/PinInput/stories/PinInputWithNoCopyPaste.storyfile.tsx?raw"
+    },
+    {
+      "name": "./src/inputs/PinInput/stories/PinInputWithValidation.storyfile.tsx"
+    },
+    {
+      "name": "./src/inputs/PinInput/stories/PinInputWithValidation.storyfile.tsx?raw"
+    }
+  ],
+  "reasons": [
+    {
+      "moduleName": "./src/ lazy ^\\.\\/.*$ include: (?%21.*node_modules)(?:\\/src(?:\\/(?%21\\.)(?:(?:(?%21(?:^%7C\\/)\\.).)*?)\\/%7C\\/%7C$)(?%21\\.)(?=.)[^/]*?\\.stories\\.(js%7Cjsx%7Cts%7Ctsx))$ chunkName: [request] namespace object"
+    }
+  ]
+}
+```
+
+Under `reasons`, you'll see a `moduleName` that looks like a path pattern. If you were to look at the non-trimmed file, you'd see the `issuerPath`'s are `"./storybook-config-entry.js"` and `"./storybook-stories.js"` because the dependency is the project's Storybook config:
+
+Under "reasons," you'll notice a `moduleName` resembling a path pattern. In the full, untrimmed file, the `issuerPath`'s are `"./storybook-config-entry.js"` and `"./storybook-stories.js"`. That's because the dependency is related to this project's Storybook configuration:
+
+```js title=".storybook/main.js"
+/** @type { import('@storybook/react-webpack5').StorybookConfig } */
+const config = {
+  stories: [
+    {
+      directory: "../",
+      files: "**/*.@(mdx|stories.@(js|jsx|ts|tsx))",
+    },
+  ],
+  // ...
+};
+export default config;
+```
+
+### Use the trimmed stats file to trace dependencies
+
+If you're trying to understand why a story file is being detected as changed, you can [search for its file path](/docs/turbosnap/troubleshooting/#why-are-no-changes-being-detected) in the trimmed stats file and trace its dependencies to see which modules were affected.
+
+## Tracing dependencies with Vite
+
+Before Storybook 8, Vite support for TurboSnap was available through the [vite-plugin-turbosnap](https://github.com/IanVS/vite-plugin-turbosnap). Starting with version 8.0, this plugin has been integrated into Storybook, so TurboSnap now supports Vite natively.
+
+With Storybook 8.0+, if Chromatic builds your Storybook for you, it will generate the stats file for you. But if you provide a prebuilt Storybook to Chromatic then you'll need add `--stats-json` to the `build-storybook` command.
+
+Whether using the plugin or Storybook 8.0+, the process is the same. When you run the `build-storybook` command, a `preview-stats.json` file is generated. This file collects information about each module being built and creates a mapping between each file and all the files that import it. It mimics the file created by Webpack but includes only the information that TurboSnap needs to perform dependency checks.
```

Comment:
@repomadness and @winkerVSbecks, I was wondering if we could use something like the following and move the `--stats-json` flag bit into an aside. Would that work for you?

```suggestion
Vite support for TurboSnap is available out of the box, starting with Storybook 8.0 and later, and does not require any additional configuration. However, if you're using an older version of Storybook, you must install the [vite-plugin-turbosnap](https://github.com/IanVS/vite-plugin-turbosnap) plugin to enable TurboSnap support with Vite. 

Similarly to Webpack, when you run the `build-storybook` command, the `preview-stats.json` file is automatically generated. It contains information about each module being built and a mapping between each file and all the imported files. This structure mirrors that used by Wepack but only includes the information TurboSnap needs to perform dependency checks. 
```

Reply:
Yea, that's a good call.